trl/models/modeling_base.py

```diff
@@ -263,7 +264,7 @@ class and the arguments that are specific to trl models. The kwargs
                                 "pytorch_model.bin.index.json",
                                 token=token,
                             )
-                        except ValueError:  # not continue training, do not have v_head weight
+                        except:  # not continue training, do not have v_head weight
```

Comment:
Can you elaborate why this change is needed?

Reply:
Hi @younesbelkada, thanks for getting back! This change has two reasons:

* At least in huggingface-hub 0.10.1, the error for "not found" is huggingface_hub.utils._errors.EntryNotFoundError, not a ValueError. This makes it fail to catch hf_hub_download() error when an entry is not found.
* This is also consistent with the code above (when trying to catch a similar scenario): https://github.com/huggingface/trl/pull/649/files#diff-be4f28845d341065eb254da5db9fa8fee2e6b5347071ef98448f27acd70fe8a7L166

Reply:
Thanks, that makes sense, would you be happy to import `EntryNotFoundError` from huggingface_hub similarly as here: https://github.com/huggingface/transformers/blob/main/src/transformers/utils/hub.py#L43 and replace the bare excepts with `except EntryNotFoundError:`? 🙏

Reply:
Hi @younesbelkada, I am thinking this would be a little tricky, as you might notice, there are [many error codes](https://github.com/huggingface/transformers/blob/6b82d936d49956ba7b43c5ee590f4868de373b65/src/transformers/utils/hub.py#L47C25-L47C25), and the logics / handler is the same for any of them? So bare excepts will be reasonable in this case? Or we can just use the general `Exception` form?

Reply:
okay, I see only two error codes are sufficient for the purpose. I have made the proposed changes and applied it to other places in that file as well. Please review it in my most recent commit.